File: 1146-greatest-common-divisor-of-strings/greatest-common-divisor-of-strings.cpp
class Solution {
public:
    string gcdOfStrings(string str1, string str2) {
        if(str1 + str2 != str2 + str1) return "";

        int gcd1 = gcd(str1.size(), str2.size());

        return str1.substr(0, gcd1);
    }

    int gcd(int a, int b) {
        if (b == 0) return a;
        return gcd(b, a % b);
    }
};

File: 1146-greatest-common-divisor-of-strings/greatest-common-divisor-of-strings.go
func gcdOfStrings(str1 string, str2 string) string {
    if str1 + str2 != str2 + str1 {
        return ""
    }

    var gcd1 int = gcd(len(str1), len(str2))

    return str1[:gcd1];
}

func gcd(a int, b int) int {
    if(b == 0) {
        return a;
    }

    return gcd(b, a % b)
}